From 67625f6748b03bb7db865933a6c441ce727fb5c2 Mon Sep 17 00:00:00 2001
From: Matthieu Herrb <matthieu@herrb.eu>
Date: Sat, 13 Oct 2018 11:51:38 +0200
Subject: [PATCH] Disable -logfile when running with elevated privileges.

An unprivileged was able to overwrite arbitrary files
in directories in which it is able to chdir, potentially
leading to privilege elevation.

CVE-2018-14665

Issue reported by Narendra Shinde

Signed-off-by: Matthieu Herrb <matthieu@herrb.eu>
---
 hw/xfree86/common/xf86Init.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hw/xfree86/common/xf86Init.c b/hw/xfree86/common/xf86Init.c
index a544b6543..6548abd33 100644
--- a/hw/xfree86/common/xf86Init.c
+++ b/hw/xfree86/common/xf86Init.c
@@ -1141,7 +1141,9 @@ ddxProcessArgument(int argc, char **argv, int i)
     }
     if (!strcmp(argv[i], "-logfile")) {
         CHECK_FOR_REQUIRED_ARGUMENT();
-        xf86CheckPrivs(argv[i], argv[i + 1]);
+        if (xf86PrivsElevated())
+              FatalError("\nInvalid argument -logfile "
+                "with elevated privileges\n");
         xf86LogFile = argv[i + 1];
         xf86LogFileFrom = X_CMDLINE;
         return 2;
-- 
2.19.1

